experimental/packages/otlp-exporter-base/src/platform/browser/util.ts

```diff
@@ -54,14 +58,17 @@ export function sendWithBeacon(
  * @param body
  * @param url
  * @param headers
+ * @param exporterTimeout
  * @param onSuccess
  * @param onError
+ * @param compressionAlgorithm
  */
 export function sendWithXhr(
   body: string | Blob,
   url: string,
   headers: Record<string, string>,
   exporterTimeout: number,
+  compressionAlgorithm: CompressionAlgorithm,
```

Comment:
This function appears to be exported from the package so this would be a breaking change of the public API.

Reply:
Also technically, you "can" also compress the payload when using `sendBeacon` with a modern browser.

Maybe consider having a new send compressed function which takes a "config" object (so it can be added to later if required without breaking the public API) and as part of this you could specify whether you want to send via XHR, sendBeacon or maybe fetch???

Reply:
Any suggestions here? it is a common practice to keep the callback parameters at the end, so if I keep the new parameter at the end with a default value, I'd be violating this practice.

Reply:
This function is called only in `otlp-proto-exporter-base` and I have modified the call there. I will wait and see if there are other thoughts if this is sufficient.

Reply:
Just noticed I missed your second comment above, I will consider it.

Reply:
I'm okay with both approaches.

A breaking change should be okay as
- we document this package to be internal-use-only, so users should not use anything exported from this package directly.
- we pin the dependency for internal consumers (`otlp-proto-exporter-base`), so changing it does not break anything for us.

If going for a breaking change, please document it accordingly in the changelog.

Reply:
> Also technically, you "can" also compress the payload when using sendBeacon with a modern browser.

@MSNev sendBeacon API doesn't take headers so there's no way to send the Content-Encoding header. Can you send me some reference if there's a newer API for sendBeacon supported in modern browsers?

Reply:
`sendBeacon` doesn't take it directly but you can pass it [via a Blob options](https://developer.mozilla.org/en-US/docs/Web/API/Blob/Blob#options)
```ts
  const headers = {
    type: 'application/json',
  };
  const blob = new Blob([JSON.stringify(body)], headers);
  navigator.sendBeacon('url', blob);
```

Reply:
But it is problematic if your blob size is bigger than the max size that can be sent.

And I've not actually tried this so there may be other complications about passing the compressed "blob" as an iterable `blobParts` that I'm not aware of.

Reply:
The blob options only takes content type, no other headers can be sent including content-encoding.

Reply:
I have made all the changes and added the changelog entry noting the breaking change, please take a look. @pichlermarc there seems to be a build issue with the lint check, can you please take a look?
@MSNev please let me know  if you think there is still a way to send the content-encoding header with sendBeacon, otherwise we can skip the compression support for it. 
Thanks.

Reply:
@MSNev can you please take a look at the above comment when you get a chance? Thanks.

Reply:
> @pichlermarc there seems to be a build issue with the lint check, can you please take a look? 

Looks like some change in this PR requires more memory for generating docs, we were close to the limit already before. We can't update dependencies (later typedoc versions require a later typescript version). 

What we can do, however, increase the max old space size. Please try to set `--max-old-space-size=8192` [here](https://github.com/open-telemetry/opentelemetry-js/blob/6be10fbd8de2a6d50cd1442379a95c7ad52f8b03/.github/workflows/lint.yml#L51).

Reply:
I think for now lets just move forward with the XHR, I did come across someone posting about using compression with `sendBeacon`, but it was not an obvious usage, I do recall that they used headers on the blob -- but I've not found the article again.

Reply:
@pichlermarc lint check passes now, but 1 unit tests  build is still failing. Do you know what could be the reason?

Reply:
@scheler I don't see an obvious reason. It seems though that the request passed to the function in `server.respondWith()` in the tests is undefined. Not sure which changes affected this.

Reply:
@pichlermarc I was able to address the issue. Can you take a look?